### PR TITLE
Add `updateEditableParams` in Transaction Controller

### PR DESCRIPTION
**Pull request by vinistevam**
## Explanation
This PR adds an `updateEditableParams` method which allows the clients to update specific properties in the `TransactionParams`. 

## References

#Fixes (task will be converted to issue soon)

### `@metamask/transaction-controller`

- **ADDED**: `updateEditableParams`  method to allow clients to update specific transaction parameters. 

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/types.ts [hunk 1]
```diff
@@ -645,6 +645,41 @@ export interface TransactionParams {
   type?: string;
 }
 
+/**
+ * Properties that can be updated in a transaction.
+ */
+export type EditableTransactionParams = {
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/types.ts:651]: Do we need this type? I fear we have so many types already that it's adding complexity if we add additional ones that are only used on single methods, especially the update methods as I don't think any others have specific types.

File: packages/transaction-controller/src/TransactionController.ts [hunk 2]
```diff
@@ -1375,6 +1376,61 @@ export class TransactionController extends BaseController<
...
+        `Cannot update editable params as no transaction metadata found`,
+      );
+    }
+
+    validateIfTransactionUnapproved(transactionMeta, 'updateEditableParams');
+
+    const editableParams: Partial<TransactionMeta> = {
+      txParams: {
+        data,
+        from: from ?? transactionMeta.txParams.from,
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:1408]: Do we need any fallbacks here?

My understanding was `merge` does a recursive merge meaning it should leave the existing value if not overridden?

File: packages/transaction-controller/src/TransactionController.ts [hunk 3]
```diff
@@ -1375,6 +1376,61 @@ export class TransactionController extends BaseController<
...
+        to,
+        value,
+        gas,
+        gasPrice,
+      },
+    };
+    editableParams.txParams = pickBy(
+      editableParams.txParams,
+      (prop) => prop !== undefined,
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:1417]: Do we need this argument, I think the default is to remove any `undefined` values?

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 4]
```diff
@@ -3819,4 +3820,82 @@ describe('TransactionController', () => {
...
+    const params: EditableTransactionParams = {
+      data: '0x0',
+      from: ACCOUNT_2_MOCK,
+      gas: '0x0',
+      gasPrice: '0x50fd51da',
+      to: ACCOUNT_MOCK,
+      value: '0x0',
+    };
+
+    // const existingTransactionMeta: TransactionMeta = {
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply packages/transaction-controller/src/TransactionController.test.ts]: Sorry, removed it.

File: packages/transaction-controller/src/TransactionController.ts (hunk 3 above)
Comment (vinistevam) [reply packages/transaction-controller/src/TransactionController.ts]: yes, I just removed it for simplicity https://github.com/MetaMask/core/pull/2056/commits/87b3ecbb7b2e039a3271ca9385cd176c1192aa18.

File: packages/transaction-controller/src/TransactionController.ts (hunk 2 above)
Comment (vinistevam) [reply packages/transaction-controller/src/TransactionController.ts]: You are right it was complaining about the type due to the fact `from` property is required in `TransactionParams`. Fix pushed.

**Pull request merged**